src/Controller/Admin/MiscController.php

```diff
@@ -253,16 +253,16 @@ public function getValidFilenameAction(Request $request): JsonResponse
      *
      * @return JsonResponse
      */
-    public function maintenanceAction(Request $request): JsonResponse
+    public function maintenanceAction(Request $request, Tool\MaintenanceModeHelperInterface $maintenanceModeHelper): JsonResponse
     {
         $this->checkPermission('maintenance_mode');
 
         if ($request->get('activate')) {
-            Tool\Admin::activateMaintenanceMode($request->getSession()->getId());
+            $maintenanceModeHelper->activate($request->getSession()->getId());
         }
 
         if ($request->get('deactivate')) {
-            Tool\Admin::deactivateMaintenanceMode();
+            $maintenanceModeHelper->deactivate();
```

Comment:
```suggestion
            if (Tool\Admin::isInMaintenanceMode()){
               Tool\Admin::deactivateMaintenanceMode();
            }
            $maintenanceModeHelper->deactivate();
```
more edge case but without this, if  you were already in legacy maintenance mode, you would see the icon but it won't exit by clicking it
![image](https://github.com/pimcore/admin-ui-classic-bundle/assets/6014195/48cb51dc-248e-45fe-bef7-4d38f5e5dae0)
and would break script that may be calling `pimcore.helpers.deactivateMaintenance`.

Reply:
Good catch! yes, we need to check if the maintenance file exists first.